client/me/gravatar-updater/index.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import request from 'superagent';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FilePicker from 'components/file-picker';
+import FormLabel from 'components/forms/form-label';
+import { getCurrentUser } from 'state/current-user/selectors';
+import Gravatar from 'components/gravatar';
+import { localize } from 'i18n-calypso';
+import * as OAuthToken from 'lib/oauth-token';
+
+export class GravatarUpdater extends Component {
+	constructor() {
+		super();
+		this.handleOnPick = this.handleOnPick.bind( this );
+	}
+
+	static propTypes = {
+		translate: React.PropTypes.func,
+		user: PropTypes.object,
+	};
+
+	handleOnPick( files ) {
```

Comment:
In general we want all user actions that modify data to go through redux as it's our canonical data flow structure and our UI just a representation of its state. There are many benefits to this, existing and planned, like supporting uploading a gravatar while offline and queuing the action for when you do get online.

As you said, removing the coupled state from the component is very important — we may have more than one instance of a "changing the avatar" flow and we want to reflect what's going on; we could show on the main "Me" sidebar profile picture that it's being uploaded as soon as you interact with the block you are building; we may want to trigger a global notice action wherever you may be in the app when the avatar was successfully (or not successfully) uploaded.